src/app/withdraw/components/util/validate-withdraw-amount.ts

```diff
@@ -0,0 +1,107 @@
+import getEmilyLimits from "@/actions/get-emily-limits";
+import yup from "yup";
+type EmilyLimits = Awaited<ReturnType<typeof getEmilyLimits>>;
+
+const getMaxError = (maxWithdrawal: number) => {
+  return `Withdrawal exceeds current cap of ${maxWithdrawal.toLocaleString(undefined, { maximumFractionDigits: 8 })} BTC`;
+};
+function convertBTCtoSats(btcValue: number) {
+  return btcValue * 1e8;
+}
+
+function checkWithdrawalCap(
+  this: yup.TestContext<yup.AnyObject>,
+  satsValue: number,
+  emilyLimits: EmilyLimits | undefined,
+) {
+  if (
+    emilyLimits?.perWithdrawalCap &&
+    satsValue > emilyLimits.perWithdrawalCap
+  ) {
+    return this.createError({
+      path: this.path,
+      message: getMaxError(emilyLimits.perWithdrawalCap / 1e8),
+    });
+  }
+  return true;
+}
+
+function checkAvailableToWithdraw(
+  this: yup.TestContext<yup.AnyObject>,
+  satsValue: number,
+  emilyLimits: EmilyLimits | undefined,
+) {
+  if (
+    emilyLimits?.availableToWithdraw &&
+    satsValue > emilyLimits.availableToWithdraw
+  ) {
+    return this.createError({
+      path: this.path,
+      message: getMaxError(emilyLimits.availableToWithdraw / 1e8),
```

Comment:
Should we return a custom error here? To differentiate with the "fixed" cap error

Reply:
yes good question not sure tho, I think it's not important info for the user

Reply:
fixed